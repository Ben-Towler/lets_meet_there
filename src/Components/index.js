import SiteHeader from './SiteHeader/SiteHeader.component';
import SearchForm from './SearchForm/SearchForm.component';
import Trip from './Trip/Trip.component';
import Button from './Button/Button.component';
import Form from './Form/Form.component';
import Flight from './Flight/Flight.component';
import FlightSummary from './FlightSummary/FlightSummary.component';
import BookingLink from './BookingLink/BookingLink.component';
import Spinner from './Spinner/Spinner.component';
import LocationSearch from './LocationSearch/locationSearch.component.tsx';

export {
  SiteHeader,
  BookingLink,
  SearchForm,
  Trip,
  Flight,
  FlightSummary,
  Button,
  Form,
  Spinner,
  LocationSearch
}