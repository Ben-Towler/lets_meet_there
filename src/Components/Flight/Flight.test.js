import React from 'react';
import renderer from 'react-test-renderer';
import { Flight } from 'Components';
import { shallow, render, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

describe('Flight', () => {
  let flight, places, flightInDirect;

  beforeEach(() => {
    flight = {
      QuoteId: 14,
      MinPrice: 278,
      Direct: true,
      OutboundLeg: {
        CarrierIds: [
          858
        ],
        OriginId: 42414,
        DestinationId: 51631,
        DepartureDate: '2020-06-24T00:00:00'
      },
      InboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 51631,
        DestinationId: 42414,
        DepartureDate: '2020-06-27T00:00:00'
      },
      QuoteDateTime: '2020-06-22T09:18:00'
    }
    flightInDirect = {
      QuoteId: 14,
      MinPrice: 278,
      Direct: false,
      OutboundLeg: {
        CarrierIds: [
          858
        ],
        OriginId: 42414,
        DestinationId: 51631,
        DepartureDate: '2020-06-24T00:00:00'
      },
      InboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 51631,
        DestinationId: 42414,
        DepartureDate: '2020-06-27T00:00:00'
      },
      QuoteDateTime: '2020-06-22T09:18:00'
    }
    places = {
      inbound: {
        origin: {
          PlaceId: 65441,
          IataCode: 'LCA',
          Name: 'Larnaca',
          Type: 'Station',
          SkyscannerCode: 'LCA',
          CityName: 'Larnaca',
          CityId: 'LARN',
          CountryName: 'Cyprus'
        },
        destination: {
          PlaceId: 42414,
          IataCode: 'BCN',
          Name: 'Barcelona',
          Type: 'Station',
          SkyscannerCode: 'BCN',
          CityName: 'Barcelona',
          CityId: 'BARC',
          CountryName: 'Spain'
        }
      },
      outbound: {
        origin: {
          PlaceId: 42414,
          IataCode: 'BCN',
          Name: 'Barcelona',
          Type: 'Station',
          SkyscannerCode: 'BCN',
          CityName: 'Barcelona',
          CityId: 'BARC',
          CountryName: 'Spain'
        },
        destination: {
          PlaceId: 65441,
          IataCode: 'LCA',
          Name: 'Larnaca',
          Type: 'Station',
          SkyscannerCode: 'LCA',
          CityName: 'Larnaca',
          CityId: 'LARN',
          CountryName: 'Cyprus'
        }
      }
    }
  });

  test('snapshot renders', () => {
    const component = renderer.create(
    <Flight 
      flight={flight}
      places={places}
    />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot renders (favourite)', () => {
    const component = renderer.create(
    <Flight 
      flight={flight}
      places={places}
      city='Barcelona'
      favCity='Barcelona'
    />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Flight Direct', () => {
    const component = render (
      <Flight 
        flight={flight}
        places={places}
        city='Barcelona'
        favCity='Barcelona'
      />
    )
    expect(component.text()).toMatch('Direct');
  });

  test('Flight InDirect', () => {
    const component = render (
      <Flight 
        flight={flightInDirect}
        places={places}
        city='Barcelona'
        favCity='Barcelona'
      />
    )
    expect(component.text()).toMatch('InDirect');
  });

});