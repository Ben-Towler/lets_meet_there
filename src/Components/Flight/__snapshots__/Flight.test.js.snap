// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight snapshot renders (favourite) 1`] = `
<div
  className="sc-AxheI dVkpdt"
>
  Barcelona
  Larnaca
  Larnaca-&gt;
  Barcelona
  Direct
  278
  <a
    aria-label="Book Flights"
    className="sc-AxmLO dTodBP"
    href="https://www.skyscanner.net/transport/flights/Barc/Larn/2020-06-24/2020-06-27"
    rel="noopener noreferrer"
    target="_blank"
    title="Book Flights"
  >
    Book Your Flights
  </a>
</div>
`;

exports[`Flight snapshot renders 1`] = `
<div
  className="sc-AxheI dVkpdt"
>
  Barcelona-&gt;
  Larnaca
  Larnaca-&gt;
  Barcelona
  Direct
  278
  <a
    aria-label="Book Flights"
    className="sc-AxmLO dTodBP"
    href="https://www.skyscanner.net/transport/flights/Barc/Larn/2020-06-24/2020-06-27"
    rel="noopener noreferrer"
    target="_blank"
    title="Book Flights"
  >
    Book Your Flights
  </a>
</div>
`;
